An IO exception occurred: Could not load cache for class: org.exist.util.io.FileFilterInputStreamCache

### What is the problem
> With existdb 3.6.1 on Windows 10, exist running as a service, whenever I wake my computer from hibernation, exist stops functioning giving  ERROR An IO exception occurred: Could not load cache for class: org.exist.util.io.FileFilterInputStreamCache

After restarting the service (or windows) exist works properly again.

### What did you expect
> No problems after hibernation, never had this issue before 3.6.1

### Context information
Please always add the following information
- eXist-db version : 3.6.1
- Java version: Java8u161)
- Operating system (Windows 10, latest update)
- 64 bit
- No custom changes.
- data directory copied from existdb 3.4, but this is supposed to be a binary-compatible release, is it not?

I have several of these in my log, just one example:
`2018-01-24 15:13:49,870 [qtp1872012709-201] ERROR (FilterInputStreamCacheFactory.java [instantiate]:81) - null 
java.lang.reflect.InvocationTargetException: null
	at sun.reflect.GeneratedConstructorAccessor151.newInstance(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_161]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_161]
	at org.exist.util.io.FilterInputStreamCacheFactory.instantiate(FilterInputStreamCacheFactory.java:72) [exist.jar:3.6.1]
	at org.exist.util.io.FilterInputStreamCacheFactory.getCacheInstance(FilterInputStreamCacheFactory.java:60) [exist.jar:3.6.1]
	at org.exist.xquery.value.BinaryValueFromInputStream.<init>(BinaryValueFromInputStream.java:33) [exist.jar:3.6.1]
	at org.exist.xquery.value.BinaryValueFromInputStream.getInstance(BinaryValueFromInputStream.java:45) [exist.jar:3.6.1]
	at org.exist.xquery.functions.request.GetData.eval(GetData.java:133) [exist-optional.jar:3.6.1]
	at org.exist.xquery.BasicFunction.eval(BasicFunction.java:74) [exist.jar:3.6.1]
	at org.exist.xquery.InternalFunctionCall.eval(InternalFunctionCall.java:41) [exist.jar:3.6.1]
	at org.exist.xquery.LetExpr.eval(LetExpr.java:99) [exist.jar:3.6.1]
	at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71) [exist.jar:3.6.1]
	at org.exist.xquery.PathExpr.eval(PathExpr.java:276) [exist.jar:3.6.1]
	at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71) [exist.jar:3.6.1]
	at org.exist.xquery.XQuery.execute(XQuery.java:261) [exist.jar:3.6.1]
	at org.exist.xquery.XQuery.execute(XQuery.java:185) [exist.jar:3.6.1]
	at org.exist.http.RESTServer.executeXQuery(RESTServer.java:1539) [exist-optional.jar:3.6.1]
	at org.exist.http.RESTServer.checkForXQueryTarget(RESTServer.java:1224) [exist-optional.jar:3.6.1]
	at org.exist.http.RESTServer.doPut(RESTServer.java:1025) [exist-optional.jar:3.6.1]
	at org.exist.http.servlets.EXistServlet.doPut(EXistServlet.java:146) [exist-optional.jar:3.6.1]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:710) [servlet-api-3.1.jar:3.1.0]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [servlet-api-3.1.jar:3.1.0]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841) [jetty-servlet-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:535) [jetty-servlet-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:566) [jetty-security-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473) [jetty-servlet-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:163) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:73) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.exist.http.urlrewrite.Forward.doRewrite(Forward.java:50) [exist-optional.jar:3.6.1]
	at org.exist.http.urlrewrite.XQueryURLRewrite.doRewrite(XQueryURLRewrite.java:548) [exist-optional.jar:3.6.1]
	at org.exist.http.urlrewrite.XQueryURLRewrite.service(XQueryURLRewrite.java:352) [exist-optional.jar:3.6.1]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [servlet-api-3.1.jar:3.1.0]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841) [jetty-servlet-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650) [jetty-servlet-9.4.7.v20170914.jar:9.4.7.v20170914]
	at de.betterform.agent.web.filter.XFormsFilter.doFilter(XFormsFilter.java:171) [betterform-exist-5.1-SNAPSHOT-20160615.jar:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1629) [jetty-servlet-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533) [jetty-servlet-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524) [jetty-security-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473) [jetty-servlet-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:455) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.Server.handle(Server.java:561) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:334) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251) [jetty-server-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279) [jetty-io-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:104) [jetty-io-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124) [jetty-io-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247) [jetty-util-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140) [jetty-util-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131) [jetty-util-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:243) [jetty-util-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:679) [jetty-util-9.4.7.v20170914.jar:9.4.7.v20170914]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:597) [jetty-util-9.4.7.v20170914.jar:9.4.7.v20170914]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_161]
Caused by: java.nio.file.NoSuchFileException: C:\WINDOWS\TEMP\_mmtfm_d8729f9f-9052-47a1-a901-49bcf0b73c168261318144176977362\mmtf_1516803229868395367594306022138.tmp
	at sun.nio.fs.WindowsException.translateToIOException(Unknown Source) ~[?:1.8.0_161]
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source) ~[?:1.8.0_161]
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source) ~[?:1.8.0_161]
	at sun.nio.fs.WindowsFileSystemProvider.newByteChannel(Unknown Source) ~[?:1.8.0_161]
	at java.nio.file.Files.newByteChannel(Unknown Source) ~[?:1.8.0_161]
	at java.nio.file.Files.createFile(Unknown Source) ~[?:1.8.0_161]
	at java.nio.file.TempFileHelper.create(Unknown Source) ~[?:1.8.0_161]
	at java.nio.file.TempFileHelper.createTempFile(Unknown Source) ~[?:1.8.0_161]
	at java.nio.file.Files.createTempFile(Unknown Source) ~[?:1.8.0_161]
	at org.exist.util.io.TemporaryFileManager.getTemporaryFile(TemporaryFileManager.java:95) ~[exist.jar:3.6.1]
	at org.exist.util.io.FileFilterInputStreamCache.<init>(FileFilterInputStreamCache.java:61) ~[exist.jar:3.6.1]
	at org.exist.util.io.FileFilterInputStreamCache.<init>(FileFilterInputStreamCache.java:55) ~[exist.jar:3.6.1]
	... 78 more`

--- a/tools/yajsw/conf/wrapper.conf.in
+++ b/tools/yajsw/conf/wrapper.conf.in
@@ -102,6 +102,7 @@ wrapper.java.additional.4=-Dlog4j.configurationFile=file:tools/yajsw/conf/log4j2
 
 # force headless mode
 wrapper.java.additional.5=-Djava.awt.headless=true
+wrapper.java.additional.6=-Djava.io.tmpdir=@server.dir@/tools/yajsw/tmp
 
 # Application parameters.  Add parameters as needed starting from 1
 wrapper.app.parameter.1=jetty

